problem1.py

```diff
@@ -0,0 +1,21 @@
+#timecomplexity O(n)
+#spacecomplexity O(1)
```

Comment:
O(n) Space, recursive stack space. 
Please also try this problem iteratively as well.